workshops/test/test_util.py

```diff
@@ -0,0 +1,138 @@
+# this probably needs refactored for py3
+from datetime import datetime
+from StringIO import StringIO
+from json import dumps
+
+from django.test import TestCase
+
+from ..models import Site, Event, Role, Person
+from ..util import PERSON_UPLOAD_FIELDS, upload_person_task_csv,\
+    verify_upload_person_task
+
+from .base import TestBase
+
+class UploadPersonTaskCSVTestCase(TestCase):
+
+    def compute_from_string(self, csv_str):
+        ''' wrap up buffering the raw string & parsing '''
+        csv_buf = StringIO(csv_str)
+        # compute & return
+        return upload_person_task_csv(csv_buf)
+
+    def test_basic_parsing(self):
+        ''' See views.PERSON_UPLOAD_FIELDS for field ordering '''
+        csv = """personal,middle,family,email\njohn,a,doe,johndoe@email.com\n,jane,a,doe,janedoe@email.com"""
+        person_tasks = self.compute_from_string(csv)
+
+        # assert
+        self.assertEqual(len(person_tasks), 2)
+
+        person = person_tasks[0]
+        for key in ('person', 'role', 'event', 'errors'):
+            self.assertIn(key, person)
+
+        person_dict = person['person']
+        for key in PERSON_UPLOAD_FIELDS:
+            self.assertIn(key, person_dict)
+
+    def test_empty_field(self):
+        ''' Ensure we don't mis-order fields given blank data '''
+        csv = """personal,middle,family,email\njohn,,doe,johndoe@email.com"""
+        person = self.compute_from_string(csv)[0]
+
+        self.assertEqual(person['person']['middle'], '')
+
+    def test_serializability_of_parsed(self):
+        csv = """personal,middle,family,email\njohn,a,doe,johndoe@email.com\n,jane,a,doe,janedoe@email.com"""
+        person_tasks = self.compute_from_string(csv)
+
+        try:
+            dumps(person_tasks)
+        except Exception:
+            self.fail('Dumping person_tasks to JSON unexpectedly failed!')
+
```

Comment:
As I mentioned in https://github.com/swcarpentry/amy/pull/165/files#r23504978, we need to test against the regression with malformed files being uploaded.  It's enough to test for CSV files that have columns with wrong names.